src/server.py

```diff
@@ -78,6 +81,37 @@ def get_date_range(dates):
                 'endDate': max(dates).strftime('%Y-%m-%d %H:%M')}
 
 
+def valve(head_rev, pushlog_id):
+    """Determine which kind of job should been returned"""
+    low_job_return = True
+    COUNTER.increase_the_counter()
+    request_list = []
+    try:
+        request_list = session.query(TaskRequests.head_rev, TaskRequests.pushlog_id,
+                                     TaskRequests.low_job_return).all()
+    except Exception:
+        session.rollback()
```

Comment:
Ok, I got that! I will add a decrease_the_counter() and decrease the counter when we catch a exception and rollback. :)